### PR TITLE
Add --local and --localPort flags to client

### PR DESCRIPTION
**Pull request by mattthaber**
Adds support for the client to invoke a local container instead of a lambda function. This allows easy local debugging without needing to package up the request your self and doing manual curls to a local port.

Adds two arguments:
`--local`: Enables Local mode. Overrides --function. Changes the invoke from an lambda invoke call to a post request to localhost
`--localPort`: localhost Port to send request too. default is 9000

QA: 

## test local invoke
- in a side terminal launch the service as a container. mount the rep in as the code base, as this branch fixes a screenshot bug in service.
```
# Correct path to your paths

docker run -p 9000:8080 -v /Users/matthaber/opensource/sanity-runner/service/src:/var/task/src ghcr.io/tophat/sanity-runner-service:2.2.0
```
In main terminal now, install and package client from thsi branch
```
make -C client install
make -C client package
```
Run a test via the example repo
```
./client/bin/sanity-runner-client-macos --test-dir example/repo/sanities/ --local --output-dir output --include google-fail-example
```

## test port change 
- launch container with different port as 9000
```
docker run -p 9001:8080 -v /Users/matthaber/opensource/sanity-runner/service/src:/var/task/src ghcr.io/tophat/sanity-runner-service:2.2.0

```
- invoke it like before but use `--localPort` to test new port
```
./client/bin/sanity-runner-client-macos --test-dir example/repo/sanities/ --local --output-dir output --include google-fail-example --localPort 9001

```

## test normal invoke works
(for internal TH)
- update the makefile in thm-sanity so the sanity-runner command uses the newly packaged one used above
```
make run-tests
```
- run tests like normal

ex:
makefile
```
.PHONY: run-test
run-test: install clean-webpack webpack upload-attachments
        echo "Running ${test} on Lambda"
                /Users/matthaber/opensource/sanity-runner/client/bin/sanity-runner-client-macos $(ARGS) --include "${test}"
```

then 
```
make run-test test=healthcheck
```

### CODE REVIEW

File: client/src/run-tests.js [hunk 1]
```diff
@@ -126,7 +188,7 @@ async function runTests(
 function _archiveTestResults(outputDir, testResults) {
     Object.entries(testResults).forEach(([key, value]) => {
         const outputPath = path.join(outputDir, key)
-        fs.outputFileSync(outputPath, value, 'utf8')
+        fs.outputFileSync(outputPath, JSON.stringify(value), 'utf8')
```
Comment (eastenluis) [inline client/src/run-tests.js:191]: Is the result not just string anymore?

Comment (mattthaber) [reply client/src/run-tests.js]: fixed

File: service/src/alertOnResult.js [hunk 2]
```diff
@@ -213,8 +213,10 @@ const constructMessage = async function(
 
     //Attachments
     const screenShots = []
-    for (const screenshotTitle of Object.keys(results.screenshots)) {
-        screenShots.push(results.screenshots[screenshotTitle])
+    if (results.screenshots) {
+        for (const screenshotTitle of Object.keys(results.screenshots)) {
```
Comment (lime-green) [inline service/src/alertOnResult.js:217]: nb: `screenShots.concat(Object.values(results.screenshots))`

**Review by lime-green — APPROVED** [review]
lgtmqa

**Pull request merged**